rubix-spi/src/main/java/com/qubole/rubix/spi/ClusterManager.java

```diff
@@ -53,10 +46,11 @@
 {
   private static Log log = LogFactory.getLog(ClusterManager.class);
 
-  private int currentNodeIndex = -1;
+  private String currentNodeName = "";
```

Comment:
why not keep `null` as default?

Reply:
yeah, reverting it.

Reply:
still initialised. Why not keep `null`?

Reply:
To avoid null check and only use empty string checks, I have done that.

Reply:
nulls are clearer to assert on instead of empty strings. You are doing a check anyways (empty string check) so we aren't saving on that. Having null ensures that even if somehow the value slips by we will stop at the usage of it instead of taking any wrong decisions.